### PR TITLE
Hide Zombie Logging

### PR DESCRIPTION
**Pull request by maxyeg**
For review.
Not sure about the names. 
And I can't pass 'nginx-bogus-logging' option to beaker(maybe because of windows).

Intended to fix Issue #3550

### CODE REVIEW

File: core/src/main/java/com/twosigma/beaker/core/Main.java [hunk 1]
```diff
@@ -96,6 +96,7 @@ private static CommandLine parseCommandLine(String[] args)
     opts.addOption(null, "require-password", false, "Ask for password when connecting");
     opts.addOption(null, "listen-interface", true, "Interface to listen on - requires ip address or '*'");
     opts.addOption(null, "portable", false, "Configuration and runtime files located in application instead of user home directory.");
+    opts.addOption(null, "nginx-bogus-logging", false, "Switchs off logging unsuccessfull requests from outdated beaker pages.");
```
Comment (scottdraves) [inline core/src/main/java/com/twosigma/beaker/core/Main.java]: nginx-bogus-logging -> hide-zombie-logging

default should be true.

Switchs off logging unsuccessfull requests from outdated beaker pages.
->
Prevent distracting logging by Beaker clients of previous server instances.

**Comment by scottdraves** [comment]
i tried bypassing the option and just forced that conditional to true, and then opened chartingTutorial.bkr, and it does not load.  i think the rule is catching too much.

File: core/src/main/java/com/twosigma/beaker/core/rest/PluginServiceLocatorRest.java [hunk 2]
```diff
@@ -727,6 +737,12 @@ else if (nginxRule.equals("ipython2"))
     }
     nginxConfig = nginxConfig.replace("%(plugin_section)s", pluginSection.toString());
     nginxConfig = nginxConfig.replace("%(extra_rules)s", this.nginxExtraRules);
+    if (this.hideZombieLogging) {
+      nginxConfig = nginxConfig.replace("%(catch_outdated_requests_rule)s",
+          this.CATCH_OUTDATED_REQUESTS_RULE);
+    } else {
+      nginxConfig = nginxConfig.replace("%(catch_outdated_requests_rule)s", "");
+    }
```
Comment (scottdraves) [inline core/src/main/java/com/twosigma/beaker/core/rest/PluginServiceLocatorRest.java]: better style:

nginxConfig = nginxConfig.replace("%(catch_outdated_requests_rule)s", this.hideZombieLogging ? this.CATCH_OUTDATED_REQUESTS_RULE : "");

**Comment by scottdraves** [comment]
OK the rule is good. :)

but it should be enabled by default.  so the cmd line flag should be --show-zombie-logging to disable this feature.

**Comment by maxyeg** [comment]
I've passed "show-zombie-logging" and it was treated by parser as an argument, but not an option. So hasOption was always false.
I've debugged GnuParser a bit and it expects options to be prefixed with '-' or '--'. In both cases it will be treated as option. And so hasOption will work.

**Comment by scottdraves** [comment]
:+1:

**Pull request merged**